### PR TITLE
LibJS: Implement a somewhat sane CreateBuiltinFunction

### PR DESCRIPTION
**Pull request by linusg**
Apparently doesn't fix or break any test262 tests, even though plenty of things were incorrect/missing before. :thinking:

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/NativeFunction.h [hunk 1]
```diff
@@ -1,12 +1,15 @@
 /*
  * Copyright (c) 2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021-2022, Linus Groh <linusg@serenityos.org>
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/NativeFunction.h:3]: Should maybe just be `2022`? You probably touched the file last year so I don't think it really matters

File: Userland/Libraries/LibJS/Runtime/PromisePrototype.cpp [hunk 2]
```diff
@@ -112,21 +111,23 @@ JS_DEFINE_NATIVE_FUNCTION(PromisePrototype::finally)
             auto* promise = TRY(promise_resolve(global_object, constructor, result));
 
             // iii. Let returnValue be a new Abstract Closure with no parameters that captures value and performs the following steps when called:
-            // iv. Let valueThunk be ! CreateBuiltinFunction(returnValue, 0, "", « »).
-            auto* value_thunk = NativeFunction::create(global_object, "", [value](auto&, auto&) -> ThrowCompletionOr<Value> {
+            auto return_value = [value](auto&, auto&) -> ThrowCompletionOr<Value> {
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/PromisePrototype.cpp:114]: (I know it is not completely related to this PR so feel free to ignore it but) Should this `value` not be in a handle?
Also a similar question for `reason` below (line 142)

**Review by davidot — APPROVED** [review]
Nice clean up.
Code looks good to me, just one tiny nitpick and a question.

File: Userland/Libraries/LibJS/Runtime/NativeFunction.h (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/NativeFunction.h:3]: Yeah, I worked on it a bit last year, so just 2022 seems odd. :^)

File: Userland/Libraries/LibJS/Runtime/PromisePrototype.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/PromisePrototype.cpp]: Great catch! Added another commit to fix this.

File: Userland/Libraries/LibJS/Runtime/NativeFunction.h (hunk 1 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Runtime/NativeFunction.h:3]: Don't want `2022` to get lonely

**Pull request merged**